src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button, Stack, Typography } from '@mui/material';
+import {
+  Component,
+  ComponentType,
+  PropsWithChildren,
+  PropsWithRef,
+  ReactNode
+} from 'react';
+import {
+  //NavigateProps,
+  NavigateFunction,
+  Params,
+  useLocation,
+  useNavigate,
+  useParams,
+  Location
+} from 'react-router';
+import { AppHistory, useAppHistory } from '../hooks/useAppHistory';
+
+interface RouterProps {
+  navigate: NavigateFunction;
+  readonly params: Params<string>;
+  location: Location;
+}
+
+type WithRouterProps<T> = T & RouterProps;
+type OmitRouter<T> = Omit<T, keyof RouterProps>;
+
+function withRouter<T>(Component: ComponentType<OmitRouter<T> & RouterProps>) {
+  return (props: OmitRouter<T>) => {
+    const location = useLocation();
+    const navigate = useNavigate();
+    const params = useParams();
+    const appHistory = useAppHistory();
+    return (
+      <Component
+        location={location}
+        navigate={navigate}
+        params={params}
+        appHistory={appHistory}
+        {...props}
+      />
+    );
+  };
+}
+
+type ErrorBoundaryState = {
+  error: Error | null;
+};
+
+type ErrorBoundaryProps = {
+  message?: string;
+  children?: ReactNode;
+  appHistory?: AppHistory;
+};
+
+class ErrorBoundaryComponent extends Component<
+  WithRouterProps<PropsWithRef<PropsWithChildren<ErrorBoundaryProps>>>,
+  ErrorBoundaryState
+> {
+  constructor(props) {
+    super(props);
+    this.state = { error: null };
+  }
+
+  componentDidCatch(error) {
+    // Catch errors in any components below and re-render with error message
+    this.setState({
+      error: error
+    });
+    // You can also log error messages to an error reporting service here
+  }
+
+  handleBackClick = () => {
+    this.props.appHistory?.goBack();
```

Comment:
`navigate = useNavigation()

navigate(-1)`

Is also possible to go the previous page.

Reply:
When the app is broken, and url is changed through React-Router, a new search is not initiated. 
This helps us maintain a custom history stack with `url`s included (`react-router` does not give us access to the history stack directly and `url`s are not accessible)